core/tests/tests.go

```diff
@@ -275,8 +275,14 @@ func setupToolCalls(bifrost *bifrost.Bifrost, config TestConfig, ctx context.Con
 				fmt.Printf("\nError in %s tool call request %d: %v\n", config.Provider, index+1, err.Error.Message)
 			} else {
 				if result.Choices[0].Message.ToolCalls != nil && len(*result.Choices[0].Message.ToolCalls) > 0 {
-					toolCall := *result.Choices[0].Message.ToolCalls
-					fmt.Printf("\n🐒 %s Tool Call Result %d: %s\n", config.Provider, index+1, toolCall[0].Function.Arguments)
+					for i, choice := range result.Choices {
+						if choice.Message.ToolCalls != nil && len(*choice.Message.ToolCalls) > 0 {
+							toolCall := *choice.Message.ToolCalls
+							fmt.Printf("\n🐒 %s Tool Call Result %d (Choice %d): %s\n", config.Provider, index+1, i+1, toolCall[0].Function.Arguments)
```

Comment:
remove all these from the codebase - fmt.Printf is an expensive line

Reply:
its only there in the test files so do we need to remove it? cause anyway it is not interfering with the core working

Reply:
use log.Println